### PR TITLE
New support for Application Migration Service (MGN) resources #1024

### PR DESCRIPTION
**Pull request by joelee**
Removal of all AWS Application Migration Service (MGN) Source Servers and Migration Jobs as requested on #1024

### CODE REVIEW

**Comment by der-eismann** [comment]
Hey @joelee, thanks for the PR! However I would kindly ask you to replace `Mgn` with `MGN` and add `Properties()` functions for the resources, since these can hold more data than the `String()` ([example](https://github.com/rebuy-de/aws-nuke/blob/main/resources/cloudfront-distributions.go#L65-L73)).

**Review by der-eismann — DISMISSED** [review]

File: resources/mgn-jobs.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		JobID: f.jobID,
+	})
+
+	return err
+}
+
+func (f *MGNJob) Properties() types.Properties {
+	properties := types.NewProperties()
+	properties.Set("JobID", f.jobID)
+	properties.Set("Arn", f.arn)
```
Comment (bjoernhaeuser) [inline resources/mgn-jobs.go:66]:
```suggestion
	properties.Set("ARN", f.arn)
```

We are unfortunately not consistent with this, though I think `ARN` is the correct one.

File: resources/mgn-source_servers.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		SourceServerID: f.sourceServerID,
+	})
+
+	return err
+}
+
+func (f *MGNSourceServer) Properties() types.Properties {
+	properties := types.NewProperties()
+	properties.Set("SourceServerID", f.sourceServerID)
+	properties.Set("Arn", f.arn)
```
Comment (bjoernhaeuser) [inline resources/mgn-source_servers.go:66]:
```suggestion
	properties.Set("ARN", f.arn)
```

Similar thing here.

File: resources/mgn-jobs.go (hunk 1 above)
Comment (joelee) [reply resources/mgn-jobs.go]: Correction made. Thank you for the review.

**Review by bjoernhaeuser — APPROVED** [review]

**Review by der-eismann — APPROVED** [review]

**Pull request merged**